test/e2e/vsphere_volume_placement.go

```diff
@@ -19,218 +19,412 @@ package e2e
 import (
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/uuid"
 	"k8s.io/kubernetes/pkg/api/v1"
 	"k8s.io/kubernetes/pkg/client/clientset_generated/clientset"
 	vsphere "k8s.io/kubernetes/pkg/cloudprovider/providers/vsphere"
 	"k8s.io/kubernetes/test/e2e/framework"
+	"os"
+	"strconv"
 )
 
 var _ = framework.KubeDescribe("Volume Placement [Feature:Volume]", func() {
 	f := framework.NewDefaultFramework("volume-placement")
 	var (
 		c                  clientset.Interface
 		ns                 string
-		vsp                *vsphere.VSphere
-		volumePath         string
 		node1Name          string
-		node1LabelValue    string
 		node1KeyValueLabel map[string]string
-
 		node2Name          string
-		node2LabelValue    string
 		node2KeyValueLabel map[string]string
-
-		isNodeLabeled bool
+		testSeup           bool
```

Comment:
The need is because when test is executed using parent `KubeDescribe` `--ginkgo.focus=Volume\sPlacement `, outer `BeforeEach` is executed, but when we need to run individual test from inner `KubeDescribe` using `--ginkgo.focus=test\sback\sto\sback\spod\screation\sand\sdeletion\swith\sdifferent\svolume\ssources\son\sthe\ssame\sworker\snode`, `BeforeEach` from parent `KubeDescribe` is not getting invoked.

so added `testSeup` flag , to control invocation of testSetupVolumePlacement, in innner `BeforeEach`